### DISCUSSION TITLE
script: parsing json by list or array

### DISCUSSION THREAD

**Discussion by 2000ede** [arendst/Tasmota#20383] [Q&A]
Hi,

If the json output is attributes, I can parse it well:
Example:
variable=total#Power#d

Unfortunately, this doesn't work if the part to be read is a list or array.
How can I read the power, for example?

`{"inverters":[{"serial":"112182852676","name":"hoymiles_HM_300_112182852676","order":0,"data_age":2,"poll_enabled":true,"reachable":true,"producing":true,"limit_relative":100,"limit_absolute":300,"AC":{"0":{"Power":{"v":32.5,"u":"W","d":1},"Voltage":{"v":234.1000061,"u":"V","d":1},"Current":{"v":0.140000001,"u":"A","d":2},"Power DC":{"v":34,"u":"W","d":1},"YieldDay":{"v":14,"u":"Wh","d":0},"YieldTotal":{"v":13.83600044,"u":"kWh","d":3},"Frequency":{"v":49.97000122,"u":"Hz","d":2},"PowerFactor":{"v":1,"u":"","d":3},"ReactivePower":{"v":0,"u":"var","d":1},"Efficiency":{"v":95.58823395,"u":"%","d":3}}},"DC":{"0":{"name":{"u":""},"Power":{"v":34,"u":"W","d":1},"Voltage":{"v":33.20000076,"u":"V","d":1},"Current":{"v":1.029999971,"u":"A","d":2},"YieldDay":{"v":14,"u":"Wh","d":0},"YieldTotal":{"v":13.83600044,"u":"kWh","d":3}}},"INV":{"0":{"Temperature":{"v":13.10000038,"u":"°C","d":1}}},"events":2}],"total":{"Power":{"v":32.5,"u":"W","d":1},"YieldDay":{"v":14,"u":"Wh","d":0},"YieldTotal":{"v":13.83600044,"u":"kWh","d":3}},"hints":{"time_sync":false,"radio_problem":false,"default_password":false},"vedirect":{"enabled":false,"total":{"Power":{"v":0,"u":"W","d":1},"YieldDay":{"v":0,"u":"Wh","d":0},"YieldTotal":{"v":0,"u":"kWh","d":2}}},"huawei":{"enabled":false,"Power":{"v":0,"u":"W","d":2}},"battery":{"enabled":false,"soc":{"v":0,"u":"%","d":0}},"power_meter":{"enabled":false,"Power":{"v":0,"u":"W","d":1}}}`

![grafik](https://github.com/arendst/Tasmota/assets/37264959/b21902d4-ecb1-4e55-a1b7-a3d5abf90f82)

**Comment by s-hadinger** [comment]
What do you mean with "parsing"? Rules? Scripting? Berry?

**Reply by 2000ede** [reply]
tasmota -> console -> script

Selecting information from a json response using script
